awsbestpractices/custom-retries/CustomRetries.java

```diff
@@ -0,0 +1,120 @@
+package example;
+
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.services.lambda.AWSLambda;
+import com.amazonaws.services.lambda.model.InvokeRequest;
+
+import java.io.IOException;
+
+public class Example implements RequestHandler<ScheduledEvent, Void> {
+
+    private AWSLambda lambdaClient;
+    private MyOtherClient otherClient;
+
+    // Should flag
+    public void retry() {
+        final InvokeRequest request = new InvokeRequest();
+        while (true) {
+            try {
+                return lambdaClient.invoke(request);
+            } catch (AmazonServiceException e) {
+            }
+        }
+    }
+
+    // Should flag
+    public void retryInDoWhile() {
+        final InvokeRequest request = new InvokeRequest();
+        do {
+            try {
+                return lambdaClient.invoke(request);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code makes multiple retries to AWS APIs. The SDK supports both automatic handling and customization of retries. You could improve the efficiency of the code by using the AWS SDK functionality.